From ac7436041d8f3db66e04eaf2cbd3f0a1fff541f9 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Tue, 1 Feb 2011 15:47:03 +0800
Subject: [PATCH 01/23] init: change firmware_dirs for Android-x86

---
 rootdir/ueventd.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/ueventd.rc b/rootdir/ueventd.rc
index 56e774bdc71..7d2fd943d4b 100644
--- a/rootdir/ueventd.rc
+++ b/rootdir/ueventd.rc
@@ -1,7 +1,7 @@
 import /vendor/etc/ueventd.rc
 import /odm/etc/ueventd.rc
 
-firmware_directories /etc/firmware/ /odm/firmware/ /vendor/firmware/ /firmware/image/
+firmware_directories /system/lib/firmware
 uevent_socket_rcvbuf_size 16M
 
 subsystem graphics
-- 
2.25.1

