From 9bb7add9d2548468de59aa1bc1a711041df0b4cd Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sun, 20 Feb 2022 12:36:10 -0500
Subject: [PATCH 09/25] Make compatible with vendor/gapps from AG

---
 device.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/device.mk b/device.mk
index e7df647..0283b3f 100644
--- a/device.mk
+++ b/device.mk
@@ -128,7 +128,7 @@ $(call inherit-product,frameworks/native/build/tablet-10in-xhdpi-2048-dalvik-hea
 
 # Get GMS
 GAPPS_VARIANT ?= pico
-$(call inherit-product-if-exists,$(if $(wildcard vendor/google/products/gms.mk),vendor/google/products/gms.mk,vendor/opengapps/build/opengapps-packages.mk))
+$(call inherit-product-if-exists,$(if $(wildcard vendor/google/products/gms.mk),vendor/google/products/gms.mk,vendor/gapps/common/common-vendor.mk))
 
 # Get native bridge settings
 $(call inherit-product-if-exists,$(LOCAL_PATH)/nativebridge/nativebridge.mk)
-- 
2.25.1

