From 7611b0fc3a16cbcda7236c7c4945f2620fc53e18 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Thu, 21 Mar 2019 17:27:45 +0800
Subject: [PATCH 62/89] Ignore configs subdir

---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 70580bdd352c..32f3e5a2ffa7 100644
--- a/.gitignore
+++ b/.gitignore
@@ -146,3 +146,5 @@ x509.genkey
 
 # Clang's compilation database file
 /compile_commands.json
+
+configs
-- 
2.30.2

