From 3b8180a62d2998aa6a136582ee0bd18e00dba9be Mon Sep 17 00:00:00 2001
From: youling257 <youling257@gmail.com>
Date: Sat, 12 Aug 2017 14:02:39 +0800
Subject: [PATCH 032/100] usbnet: [TEMP HACK] force the interface name to ethX

Signed-off-by: youling257 <youling257@gmail.com>
---
 drivers/net/usb/usbnet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/usb/usbnet.c b/drivers/net/usb/usbnet.c
index f4f37ecfed58..be1cc523feee 100644
--- a/drivers/net/usb/usbnet.c
+++ b/drivers/net/usb/usbnet.c
@@ -1682,7 +1682,7 @@ usbnet_probe (struct usb_interface *udev, const struct usb_device_id *prod)
 	dev->interrupt_count = 0;
 
 	dev->net = net;
-	strcpy (net->name, "usb%d");
+	strcpy (net->name, "eth%d");
 	memcpy (net->dev_addr, node_id, sizeof node_id);
 
 	/* rx and tx sides can use different message sizes;
-- 
2.17.1

