From 4bf45b1d6ffd350a946034d88b259bee2b648eea Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Thu, 21 Mar 2019 17:27:45 +0800
Subject: [PATCH 46/54] Ignore configs subdir

---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 44e34991875e..ce66f868618c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -156,3 +156,5 @@ x509.genkey
 
 # Documentation toolchain
 sphinx_*/
+
+configs
-- 
2.30.2

