From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Povilas Staniulis <wdmonster@gmail.com>
Date: Thu, 23 Mar 2017 21:33:24 +0800
Subject: [PATCH 23/54] HID: multitouch: add ids of Uiworks

---
 drivers/hid/hid-multitouch.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/hid/hid-multitouch.c b/drivers/hid/hid-multitouch.c
index 205c4526705d..0320d22e128c 100644
--- a/drivers/hid/hid-multitouch.c
+++ b/drivers/hid/hid-multitouch.c
@@ -198,6 +198,7 @@ static void mt_post_parse(struct mt_device *td, struct mt_application *app);
 #define MT_CLS_EGALAX				0x0103
 #define MT_CLS_EGALAX_SERIAL			0x0104
 #define MT_CLS_TOPSEED				0x0105
+#define MT_CLS_UIWORKS				0x0105
 #define MT_CLS_PANASONIC			0x0106
 #define MT_CLS_FLATFROG				0x0107
 #define MT_CLS_GENERALTOUCH_TWOFINGERS		0x0108
@@ -308,6 +309,9 @@ static const struct mt_class mt_classes[] = {
 		.sn_move = 4096,
 		.sn_pressure = 32,
 	},
+	{ .name = MT_CLS_UIWORKS,
+		.quirks =  MT_QUIRK_ALWAYS_VALID,
+	},
 	{ .name = MT_CLS_TOPSEED,
 		.quirks = MT_QUIRK_ALWAYS_VALID,
 		.is_indirect = true,
@@ -2076,6 +2080,16 @@ static const struct hid_device_id mt_devices[] = {
 		MT_USB_DEVICE(USB_VENDOR_ID_TOUCH_INTL,
 			USB_DEVICE_ID_TOUCH_INTL_MULTI_TOUCH) },
 
+	/* Uiworks */
+	{ .driver_data = MT_CLS_UIWORKS,
+		HID_USB_DEVICE(0x23FF, 0x1600) },
+	{ .driver_data = MT_CLS_UIWORKS,
+		HID_USB_DEVICE(0x23FF, 0x1601) },
+	{ .driver_data = MT_CLS_UIWORKS,
+		HID_USB_DEVICE(0x23FF, 0x1602) },
+	{ .driver_data = MT_CLS_UIWORKS,
+		HID_USB_DEVICE(0x23FF, 0x1603) },
+
 	/* Unitec panels */
 	{ .driver_data = MT_CLS_NSMU,
 		MT_USB_DEVICE(USB_VENDOR_ID_UNITEC,
