From a907442d8bc73155d2960ca132ab4d0c8242cff1 Mon Sep 17 00:00:00 2001
From: youling257 <youling257@gmail.com>
Date: Sat, 12 Aug 2017 14:12:41 +0800
Subject: [PATCH 034/106] shmem: enable user xattr for tmpfs

Signed-off-by: youling257 <youling257@gmail.com>
---
 mm/shmem.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mm/shmem.c b/mm/shmem.c
index 7c6b6d8f6c39..628eab7c4c85 100644
--- a/mm/shmem.c
+++ b/mm/shmem.c
@@ -3283,6 +3283,12 @@ static int shmem_xattr_handler_set(const struct xattr_handler *handler,
 	return simple_xattr_set(&info->xattrs, name, value, size, flags, NULL);
 }
 
+static const struct xattr_handler shmem_user_xattr_handler = {
+	.prefix = XATTR_USER_PREFIX,
+	.get = shmem_xattr_handler_get,
+	.set = shmem_xattr_handler_set,
+};
+
 static const struct xattr_handler shmem_security_xattr_handler = {
 	.prefix = XATTR_SECURITY_PREFIX,
 	.get = shmem_xattr_handler_get,
@@ -3300,6 +3306,7 @@ static const struct xattr_handler *shmem_xattr_handlers[] = {
 	&posix_acl_access_xattr_handler,
 	&posix_acl_default_xattr_handler,
 #endif
+	&shmem_user_xattr_handler,
 	&shmem_security_xattr_handler,
 	&shmem_trusted_xattr_handler,
 	NULL
-- 
2.17.1

