From 85ec3eeaec0cd509c2b7a49a1a3ee1da0cf56344 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Tue, 15 Dec 2020 11:42:48 -0500
Subject: [PATCH 03/11] Make sure ntfs-3g has correct permissions on init

---
 initrd/init | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/initrd/init b/initrd/init
index 0c286a0f..cd7422d8 100755
--- a/initrd/init
+++ b/initrd/init
@@ -23,6 +23,8 @@ else
 fi
 exec 2>> $LOG
 
+chmod 755 /sbin/mount.ntfs-3g
+
 # early boot
 if test x"$HAS_CTTY" != x"Yes"; then
 	# initialise /proc and /sys
-- 
2.17.1

