From c535a89cac6ba8cac599764fe3cb32e0aa0c065a Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sun, 24 Oct 2021 19:47:48 -0400
Subject: [PATCH 20/28] Add AGP apps

---
 device.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/device.mk b/device.mk
index 94243fb..ec84ba5 100644
--- a/device.mk
+++ b/device.mk
@@ -132,3 +132,7 @@ $(call inherit-product-if-exists,$(if $(wildcard vendor/google/products/gms.mk),
 $(call inherit-product-if-exists,$(LOCAL_PATH)/nativebridge/nativebridge.mk)
 
 $(call inherit-product,$(if $(wildcard $(PRODUCT_DIR)packages.mk),$(PRODUCT_DIR),$(LOCAL_PATH)/)packages.mk)
+
+# Add agp-apps
+$(call inherit-product-if-exists, vendor/prebuilts/agp-apps/agp-apps.mk)
+
-- 
2.25.1

