From 9b30d51ace3b145aaa43b2f5f68bad45a30fcc42 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Wed, 27 Oct 2021 21:21:32 -0400
Subject: [PATCH 26/28] inherit foss apps

---
 device.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/device.mk b/device.mk
index 11a2832..26998d8 100644
--- a/device.mk
+++ b/device.mk
@@ -128,6 +128,9 @@ $(call inherit-product,frameworks/native/build/tablet-10in-xhdpi-2048-dalvik-hea
 GAPPS_VARIANT ?= pico
 $(call inherit-product-if-exists,$(if $(wildcard vendor/google/products/gms.mk),vendor/google/products/gms.mk,vendor/opengapps/build/opengapps-packages.mk))
 
+# foss apps
+$(call inherit-product-if-exists, vendor/foss/foss.mk)
+
 # Get native bridge settings
 ifeq ($(patsubst %x86,,$(lastword $(TARGET_PRODUCT))),)
 $(call inherit-product-if-exists,$(LOCAL_PATH)/nativebridge/nativebridge.mk)
-- 
2.25.1

