From 677c545af72d5f242218088b300da50974dc9c58 Mon Sep 17 00:00:00 2001
From: Ronny Buchmann <60157654+ronnybuchmann@users.noreply.github.com>
Date: Mon, 5 Apr 2021 23:13:23 +0200
Subject: [PATCH 24/36] init.sh: better integer check for pcm

---
 init.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/init.sh b/init.sh
index 31b1b3d..8a8b5a0 100644
--- a/init.sh
+++ b/init.sh
@@ -74,9 +74,9 @@ function init_hal_audio()
 	
 	# choose the first connected HDMI port on card 0 or 1
 	pcm=$(alsa_ctl store -f - 0 2>/dev/null| grep "CARD" -A 2 | grep "value true" -B 1 | grep "HDMI.*pcm" | head -1 | sed -e's/.*pcm=\([0-9]*\).*/\1/')
-	[ "$pcm" -ge 0 ] 2>/dev/null && set_prop_if_empty hal.audio.out "pcmC0D${pcm}p"
+	[ -z "${pcm##*[!0-9]*}" ] || set_prop_if_empty hal.audio.out "pcmC0D${pcm}p"
 	pcm=$(alsa_ctl store -f - 1 2>/dev/null| grep "CARD" -A 2 | grep "value true" -B 1 | grep "HDMI.*pcm" | head -1 | sed -e's/.*pcm=\([0-9]*\).*/\1/')
-	[ "$pcm" -ge 0 ] 2>/dev/null && set_prop_if_empty hal.audio.out "pcmC1D${pcm}p"
+	[ -z "${pcm##*[!0-9]*}" ] || set_prop_if_empty hal.audio.out "pcmC1D${pcm}p"
 }
 
 function init_hal_bluetooth()
-- 
2.17.1

