From fde50a2c37e81472f67690cf8649d78585f56877 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Tue, 12 Sep 2017 16:22:04 +0800
Subject: [PATCH 13/22] init: don't bail out even no SELinux domain defined

---
 init/service.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/init/service.cpp b/init/service.cpp
index 69f944eb2..c06c600f9 100644
--- a/init/service.cpp
+++ b/init/service.cpp
@@ -86,11 +86,18 @@ static Result<std::string> ComputeContextFromExecutable(const std::string& servi
         free(new_con);
     }
     if (rc == 0 && computed_context == mycon.get()) {
+#if defined(__ANDROID__)
+        LOG(WARNING) << "File " << service_path << "(labeled \"" << filecon.get()
+                     << "\") has incorrect label or no domain transition from " << mycon.get()
+                     << " to another SELinux domain defined.";
+        return "skip";
+#else
         return Error() << "File " << service_path << "(labeled \"" << filecon.get()
                        << "\") has incorrect label or no domain transition from " << mycon.get()
                        << " to another SELinux domain defined. Have you configured your "
                           "service correctly? https://source.android.com/security/selinux/"
                           "device-policy#label_new_services_and_address_denials";
+#endif
     }
     if (rc < 0) {
         return Error() << "Could not get process context";
-- 
2.30.2

